src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -307,7 +307,9 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
                 ?: constructorProperties[parameterSpec.name]?.kdoc?.takeUnless { it.isEmpty() }
         if (kdoc != null) {
           if (isNotEmpty() && index == 0) add("\n")
-          add("@param %L %L", parameterSpec.name, kdoc.ensureEndsWithNewLine())
+          val isProperty = parameterSpec.name in constructorProperties
+          val tag = if (isProperty) "@property" else "@param"
+          add("$tag %L %L", parameterSpec.name, kdoc.ensureEndsWithNewLine())
```

Comment:
We might have to figure out the ordering of the tags in this case. According to the [Android Kotlin Style Guide](https://developer.android.com/kotlin/style-guide#documentation) `@param` should come before `@property`. Without ordering we may be mixing it up with `@param` possible coming between two `@property`?

Reply:
Interesting. My personal preference would be to use the ordering of the constructor params for the respective tags, but we should adhere to the Kotlin/Android style guides whenever possible.

Reply:
I'll merge as-is: there's no mention on the order of those tags in the official Kotlin style guide, and hardcoding specific order based on Android's style guide doesn't seem like a good idea. Making the order of the tags reflect the ordering of the things in code feels like the most meaningful approach, plus, you can make `@param`s come before `@property`s by structuring the class header to reflect that ordering. Happy to change this later based no the feedback.